modules/fatfs/zephyr_fatfs_config.h

```diff
@@ -66,6 +67,14 @@
 #define FF_FS_EXFAT		CONFIG_FS_FATFS_EXFAT
 #endif /* defined(CONFIG_FS_FATFS_EXFAT) */
 
+#if defined(CONFIG_FS_FATFS_REENTRANT)
+#undef FF_FS_REENTRANT
+#undef FF_FS_TIMEOUT
+#include <zephyr/kernel.h>
+#define FF_FS_REENTRANT		CONFIG_FS_FATFS_REENTRANT
+#define FF_FS_TIMEOUT		K_FOREVER
```

Comment:
I'm not familiar with the internals of how this works, but let's assume that there is one partition with a log file and any other random file, if the log file is opened at startup and let's say it moves to a new file once an hour (or even once a day) but there's a way for files to be retrieved from this file system - what happens if a client requests the "other random file", does their file access part of their code (let's assume MCUmgr) then halt for **1 hour/1 day** until the mutex can be taken? If so, that seems rather inconvenient for all users of this, and unacceptable

Reply:
The reentrant option does not lock the media upon open. It only locks file access operations so that volume access is serialized. E.g. not two threads are trying to send commands over the SPI to the external Flash/SD. You only need to wait for the other open/write/... operation is finsihed.
I added a test that demonstrates that you may still have two files opened at the same time accessing them from different threads. Hope that helps.

Reply:
> I'm not familiar with the internals of how this works, but let's assume that there is one partition with a log file and any other random file, if the log file is opened at startup and let's say it moves to a new file once an hour (or even once a day) but there's a way for files to be retrieved from this file system - what happens if a client requests the "other random file", does their file access part of their code (let's assume MCUmgr) then halt for **1 hour/1 day** until the mutex can be taken? If so, that seems rather inconvenient for all users of this, and unacceptable

This will not be a problem. Generally most of functions, there are exceptions, call `validate` (https://github.com/zephyrproject-rtos/fatfs/blob/master/ff.c#L3613) that does locking for the operation, and do `LEAVE_FF` to retrun that does unlocking, https://github.com/zephyrproject-rtos/fatfs/blob/427159bf95ea49b7680facffaa29ad506b42709b/ff.c#L237.